# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsGenConList.cpp
# Commit: e37f6ffb3902
# Full Hash: e37f6ffb39025dd0b9abd9096baab78c8cb2d5fe
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-06-24 09:35:42
# Description:
#   Bug 1776211 - Assertion failure: aNode (aNode cannot be nullptr!), at /layout/base/nsGenConList.h:83 r=emilio
#   
#   Specifically handle empty lists in nsGenConList::BinarySearch. This
#   was hidden because the pre-existing caller also handles empty lists before
#   calling the binary search algorithm.
# ==============================================================================

diff -r abadcb426f52 -r e37f6ffb3902 layout/base/nsGenConList.cpp
--- a/layout/base/nsGenConList.cpp	Thu Jun 23 18:59:00 2022 +0300
+++ b/layout/base/nsGenConList.cpp	Thu Jun 23 16:05:30 2022 +0000
@@ -127,6 +127,10 @@
 
 nsGenConNode* nsGenConList::BinarySearch(
     const mozilla::FunctionRef<bool(nsGenConNode*)>& aIsAfter) {
+  if (mList.isEmpty()) {
+    return nullptr;
+  }
+
   // The range of indices at which |aNode| could end up.
   // (We already know it can't be at index mSize.)
   uint32_t first = 0, last = mSize - 1;